parachain/pallets/ethereum-beacon-client/src/lib.rs

```diff
@@ -342,51 +349,69 @@ pub mod pallet {
 		}
 
 		#[pallet::call_index(3)]
-		#[pallet::weight(1000)]
+		#[pallet::weight(1_000_000_000)]
+		#[transactional]
+		pub fn block_bridge(origin: OriginFor<T>) -> DispatchResult {
+			let _sender = ensure_root(origin)?;
+
+			<Blocked<T>>::set(true);
+
+			log::info!(target: "ethereum-beacon-client","💫 bridge emergency blocked from governance.");
+
+			Self::deposit_event(Event::BridgeBlocked { blocked: true });
+
+			Ok(())
+		}
+
+		#[pallet::call_index(4)]
+		#[pallet::weight(10_000_000_000)]
 		#[transactional]
-		pub fn unblock_bridge(origin: OriginFor<T>) -> DispatchResult {
+		pub fn unblock_bridge(
+			origin: OriginFor<T>,
+			check_points: Option<Vec<CheckpointSyncOf<T>>>,
```

Comment:
I would assume not nessessary all in in one extrinsic, multiple calls each with partial of the missing data should also be fine. 
And as mentioned above, with some monitoring components running(not scope of this PR) out-of-sync can be easily checked so that we can respond in time. For normal cases our beacon relayer will just recover smoothly from breakpoint without the `backfill`  here.

Reply:
Will multiple calls mean multiple governance proposals?

Reply:
Probably. Not sure if one governance proposal which contains multiple calls(like `batch_all` but can be scheduled in multiple blocks) is possible or we'll need multiple proposals. And based on improvements introduced in [Gov2](https://medium.com/polkadot-network/gov2-polkadots-next-generation-of-decentralised-governance-4d9ef657d11b) multiple propsals/referendums can be fast-tracked and run simultaneously I would suspect it's not a big problem.

Reply:
I have an idea: The governance proposal would include the _hashes_ of the checkpoints.

The proposal would be the following extrinsic `begin_recovery`:

```rust
fn begin_recovery(checkpoints: Vec<CheckPointHash>);
```

Then any user can permissionlessly sync individual checkpoints, as long as they are in the list above:
```rust
fn sync_recovery_checkpoint(checkpoint: CheckPoint);
```

Once all checkpoints have been synced,  call this extrinsic to unblock the bridge:
```rust
fn complete_recovery();
```

Reply:
Cool, definitely more friendly for governance! I'll do some refactoring accordingly.

Reply:
@vgeddes https://github.com/Snowfork/snowbridge/pull/795/commits/3fbe0eb1af318e56b2fed9b87b0a531369bc04fa